neuro_san/internals/utils/asyncio_executor_pool_provider.py

```diff
@@ -0,0 +1,38 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+"""
+See class comments
+"""
+from leaf_common.asyncio.asyncio_executor_pool import AsyncioExecutorPool
+
+
+class AsyncioExecutorPoolProvider:
+    """
+    Class providing server-wide instance of AsyncioExecutorPool
+    for use by specific agent services.
+    """
+
```

Comment:
Yes, I know - this is a global variable.
But I don't see options that would be less ugly.

Reply:
Well, they both use the same AsyncExecutor instances, so I don't see why not technically.
But policy is another thing - we can imagine a dictionary of Executor pools where different server components
use separate pools.